src/components/FileInputField/FileInputField.jsx

```diff
@@ -99,6 +101,30 @@ export const FileInputField = React.forwardRef((props, ref) => {
     }
   };
 
+  const handleReset = useCallback((event) => {
+    setSelectedFileNames([]);
+    onFilesChanged([], event);
+  }, [onFilesChanged]);
+
+  useEffect(() => {
+    const inputEl = internalInputRef.current;
+    if (!inputEl) {
+      return;
+    }
+
+    const { form } = inputEl;
+    if (!form) {
+      return;
+    }
+
+    form.addEventListener('reset', handleReset);
+
+    // eslint-disable-next-line consistent-return
```

Comment:
Please, remove the `eslint-disable` and replace `return;` with `return () => {};` to prevent need of this rule. This is how we do it across RUI.